source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -1276,10 +1275,11 @@ void OAuth2Filter::addResponseCookies(Http::ResponseHeaderMap& headers,
   }
 }
 
-void OAuth2Filter::sendUnauthorizedResponse() {
+void OAuth2Filter::sendUnauthorizedResponse(const std::string& details) {
+  ENVOY_LOG(warn, "Responding with 401 Unauthorized. Cause: {}", details);
   config_->stats().oauth_failure_.inc();
```

Comment:
Great work. Do you also plan on adding a cause label here? I guess that might require that details values are enumerable.

Reply:
Nevermind. I don't want to risk delaying the merge of this PR. My thought was that maybe it would be posssible to add a "cause" label to `oauth_failure_.inc()`, but all oauth metrics have only generic labels like `envoy_http_conn_manager_prefix` so there's no precedent.